aten/src/ATen/native/cuda/MiscUtils.h

```diff
@@ -38,11 +45,20 @@ struct MAGMAQueue {
 
   // Destructor
   ~MAGMAQueue() {
+#if CUDA_VERSION >= 11000
+    // We've manually set the math mode to CUBLAS_DEFAULT_MATH, now we
+    // should restore the original math mode back
+    cublasHandle_t handle = magma_queue_get_cublas_handle(magma_queue_);
+    TORCH_CUDABLAS_CHECK(cublasSetMathMode(handle, original_math_mode));
```

Comment:
You can't do this: this might cause an exception to be thrown in the destructor. You can make the call and simply not check its result, instead.

However: how does this setting of cuBLAS's math mode work in a multithreaded context? If a user runs a MAGMA operation on one thread, for example, and simultaneously runs a GEMM in another thread, then is it possible that GEMM will run without TF32?

Reply:
Oh wait - is that why you're changing the handle?

Reply:
cuBLAS handles are thread_local, each thread has its own handle, so we are safe to change the math mode here.

Reply:
check removed